Cpp inclusion guards (#5992)
Codecademy/docs: content/cpp/concepts/inclusion-guards/inclusion-guards.md

---

> Title: 'inclusion-guards'

Reviewer: The title doesn't need punctuation and should be in title case.

```suggestion
Title: 'Inclusion Guards'
```

Description: 'Used in C++ to prevent multiple inclusions of the same header file.'
Subjects:

>   - 'Computer Science'

Reviewer: We can add a few more subjects from [subjects.md](https://github.com/Codecademy/docs/blob/main/documentation/subjects.md).

```suggestion
  - 'Code Foundations'
  - 'Computer Science'
```

Tags:
  - 'Error Handling'
  - 'Compilation'
CatalogContent:

>   - 'learn-c-plus-plus'

Reviewer: We can add a paid course path as well - 

```suggestion
  - 'learn-c-plus-plus'
  - 'paths/computer-science'
```

---

> **Inclusion guards**, also known as header guards, are a common technique used in C++ to prevent multiple inclusions of the same header file, which can cause errors and increase compilation time. These are preprocessor directives that ensure the file is included only once.

Reviewer: We can simplify this as follows -

```suggestion
**Inclusion guards**, also known as **header guards**, are a common technique used in C++ to prevent multiple inclusions of the same header file. Without inclusion guards, including the same header file multiple times in a project could lead to redefinition errors and increased compilation time.

These guards use preprocessor directives to ensure a header file is included only once during compilation.
```

## Syntax

```cpp
#ifndef MY_HEADER_H
#define MY_HEADER_H

// Your header file content here

#endif // MY_HEADER_H
```

## Example

> Let's say you have a header file named example.h:

Reviewer: The file name should be backticks.

```suggestion
Let's say you have a header file named `example.h`:
```

```cpp
// example.h

> #ifndef EXAMPLE_H
> #define EXAMPLE_H

Reviewer: Add comments for explanation.

```suggestion
#ifndef EXAMPLE_H  // Checks if EXAMPLE_H has been defined
#define EXAMPLE_H // Defines it and includes the header content
```

myFunction();

#endif // EXAMPLE_H
```

> And your source file is main.cpp:

Reviewer: File name should be in backticks.

```suggestion
And your source file is `main.cpp`:
```

```cpp
// main.cpp
#include "example.h"
#include "example.h" // This will be ignored due to the inclusion guard

int main() {
    myFunction();
    return 0;
}
```

> By using inclusion guards, you ensure that example.h is only included once, even if you include it multiple times in different files or within the same file.

Reviewer: File name should be in backticks.

```suggestion
By using inclusion guards, you ensure that `example.h` is only included once, even if you include it multiple times in different files or within the same file.
```
